modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py

```diff
@@ -157,41 +159,61 @@ def deploy_axis_func(
         ----------
         axis : {0, 1}
             The axis to perform the function along.
-        func : callable
-            The function to perform.
+        invocable : Invocable
+            The function to perform with its args and kwargs.
         num_splits : int
             The number of splits to return (see `split_result_of_axis_func_pandas`).
         maintain_partitioning : bool
             If True, keep the old partitioning if possible.
             If False, create a new partition layout.
         *partitions : iterable
             All partitions that make up the full axis (row or column).
-        **kwargs : dict
-            Additional keywords arguments to be passed in `func`.
+        lengths : iterable, default: None
+            The list of lengths to shuffle the partition into.
+        manual_partition : bool, default: False
+            If True, partition the result with lengths.
 
         Returns
         -------
         list
             A list of distributed.Future.
         """
-        lengths = kwargs.get("_lengths", None)
         result_num_splits = len(lengths) if lengths else num_splits
         return DaskWrapper.deploy(
```

Comment:
I think this would be a little counter-intuitive, since `deploy` is meant to return a Future. This behavior is consistent with the [`RayTask`](https://github.com/modin-project/modin/pull/4715/files#diff-c4ce653a77903b2db19f7ed67000840fcdd4ce2ec60610766df77a7cb48a7090) class, which currently also requires explicit calls to `ray.gather` on arguments being passed in.

Reply:
Actually, on further examination it seems like this problem is avoided altogether in Ray, and we just need to call `deserialize` at the call site ([here](https://github.com/modin-project/modin/blob/d04ab97e0227e2cc4c9bd46c6577c09851e94131/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py#L390)).

Reply:
`client.submit` auto-materializes all futures contained in most of python containers. `ray_func.remote` can't do that so we have to call `ray.get` inside of a worker process. However, we shouldn't do data materialization in the main thread.